components/admin_console/plugin_management/plugin_management.jsx

```diff
@@ -606,17 +608,40 @@ export default class PluginManagement extends AdminSettings {
         return this.installFromUrl(true);
     }
 
+    handleRemovePluginCancel = () => {
+        this.setState({
+            showRemoveModal: false,
+        });
+        this.resolveRemoveModal(false);
+    }
+
+    handleRemovePlugin = () => {
+        this.setState({showRemoveModal: false});
+        this.resolveRemoveModal(true);
+    }
+
+    showRemovePluginModal = () => {
+        this.setState({showRemoveModal: true});
+        return new Promise((res) => {
```

Comment:
This would not be necessary if we let the result of the modal just call the associated handler as suggested above.